src/test/java/me/karakelley/http/server/HttpRequestReaderTest.java

```diff
@@ -85,9 +91,37 @@ void testWhenABodyIsPresent() {
     assertEquals("hey", new String(request.getBody()));
   }
 
+  @Test
+  void testThatIOExceptionIsCaughtAndLogged() {
+    withAppender(logger -> {
+      ByteArrayInputStream inputStream = new ByteArrayInputStream("GET / HTTP\r\n\r\n".getBytes());
+      HttpRequestReader requestReader = new HttpRequestReader(inputStream);
+      try {
+        inputStream.close();
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
+      assertTrue(inMemoryAppender.getEvents().contains("java.io.IOException: Stream closed"));
+    });
+  }
+
   private Request parse(String request) {
     ByteArrayInputStream inputStream = new ByteArrayInputStream(request.getBytes());
     HttpRequestReader requestReader = new HttpRequestReader(inputStream);
     return requestReader.read(0);
   }
+
+  private void withAppender(Consumer<Logger> loggerConsumer) {
```

Comment:
I like that you're managing your global state by using a lambda instead of polluting your test with `try`/`finally`. I'm not sure why you need a `Consumer<Logger>` as opposed to a `Runnable`... you don't seem to use the `logger` argument above.

I feel like it could be cleaner if you implemented something like this...

```
@Test
void testThatIOExceptionIsCaughtAndLogged() {
  // Not sure if `LogEvent` is the right type here, but you get the idea...
  List<LogEvent> logEvents = withCapturedLogging(() -> {
    ByteArrayInputStream inputStream = new ByteArrayInputStream("GET / HTTP\r\n\r\n".getBytes());
    HttpRequestReader requestReader = new HttpRequestReader(inputStream);
    try {
      inputStream.close();
    } catch (IOException e) {
      e.printStackTrace();
    }
  });

  assertTrue(logEvents.contains("java.io.IOException: Stream closed"));
}

Reply:
Im still a shakey on lambdas. I've watched 2 tutorials this morning and read a few more posts and I just can't seem to write them from scratch myself. Most of the things I read/watch say you need to implement an interface which I know I don't need to do in this case. Anyway, if you have time to maybe pair on this today or walk me through lambdas that'd be awesome.